Library/Formula/mactelnet.rb

```diff
@@ -0,0 +1,318 @@
+class Mactelnet < Formula
+  homepage "https://github.com/haakonnessjoen/MAC-Telnet"
+  url "https://github.com/haakonnessjoen/MAC-Telnet/archive/v0.4.0.tar.gz"
+  sha1 "57f2a4ae4930e1b4d57c41dd2fafb38ca1142ec5"
+
+  depends_on "gettext"
+
+  def install
+    system "make", "install", "PREFIX=#{prefix}"
+  end
+
+  # MAC-Telnet 0.4.0 doesn't build against Mac OS X:
+  #   https://github.com/haakonnessjoen/MAC-Telnet/issues/13
+  #
+  # The patch below was submitted as:
+  #   https://github.com/haakonnessjoen/MAC-Telnet/pull/14
+  patch :DATA
+
+  test do
+    # There isn't really a way to test these tools without a Mikrotik device around.
+    # We can, however, check that it tries to connect and fails in the right way.
+    command = ["#{bin}/mactelnet"] + %w(-t 1 -u user -p pass 00:11:22:33:44:55)
+
+    # Set up a pipe that we read into a string
+    pipe_r, pipe_w = IO.pipe
+    pipe_w.sync = true
+
+    output = ""
+    pipe_reader = Thread.new {
+      while str = pipe_r.read(256)
+        output << str
+      end
+    }
+
+    # Run the command, with stdout/stderr connected to our pipe
+    spawn(*command,
+      :in           => :close,
+      [:out, :err]  => pipe_w,
+      )
+    pipe_w.close
+
+    # Let it finish
+    Process.wait
+    pipe_reader.join
+
+    # Did it do what it's supposed to do?
+    # Process::Status#exitstatus is nil if the process terminated
+    $?.exitstatus == 1 && output == "Connection failed.\n"
```

Comment:
Can we just run the command with `shell_output` which allows asserting the exit code and matching output with `assert_match`?